loadtest/loadtest.go

```diff
@@ -252,6 +253,12 @@ func New(config *Config, nc NewController, log *mlog.Logger) (*LoadTester, error
 		return nil, fmt.Errorf("could not validate configuration: %w", err)
 	}
 
+	if 5*config.UsersConfiguration.MaxActiveUsers/4 > syscall.RLIMIT_NOFILE {
```

Comment:
Why the 5? I am thinking the value should be `MaxActiveUsers + (MaxActiveUsers/4)`.
Also, since we are here, let's update the original values where the `500` came from:

https://github.com/mattermost/mattermost-load-test-ng/blob/fe31d56e2024672fc5c5d4400fa0045a4937a366/api/agent.go#L318-L320 

Let's make those 500 values to be `MaxActiveUsers/4` instead.

> @streamer45 I have made the changes. Can I get info on failing testcases ?

The reason the tests are failing is because `syscall.RLIMIT_NOFILE` is just a constant (its value is 7) that is used to denote what type of call to make to the underlying operating system. It's not the number we are looking for. 

To get that we need to use `syscall.GetRlimit()`, e.g.:

```go
var rlimit syscall.Rlimit
if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rlimit); err != nil {
  return nil, err
}
```

After that call `rlimit.Max` will hold the value we are interested in.

Reply:
Thanks! Now I understood the stuff